src/transformers/models/ernie4_5_moe/modeling_ernie4_5_moe.py

```diff
@@ -288,6 +287,46 @@ def forward(self, hidden_states):
         return hidden_states + self.e_score_correction_bias.squeeze()
 
 
+class Ernie4_5_MoeNaiveMoe(nn.ModuleList):
+    def __init__(self, config):
+        super().__init__()
+        self.top_k = config.num_experts_per_tok
+        self.num_experts = config.num_local_experts
+        for _ in range(self.num_experts):
+            self += [Ernie4_5_MoeMLP(config, config.moe_intermediate_size)]
+
+    def forward(self, hidden_states, e_score_correction_bias, device_type):
+        # router_logits: (batch * sequence_length, n_experts)
+        with torch.autocast(device_type=device_type, enabled=False):  # Force float32
+            routing_weights = F.softmax(hidden_states, dim=1, dtype=torch.float)
```

Comment:
```suggestion
            routing_weights = F.softmax(routing_logits, dim=1, dtype=torch.float)
```
We need the logits from the gate here, if we move things